views/calendar.erb

```diff
@@ -0,0 +1,219 @@
+   <h2>Calendar</h2>
+   <details class="" style="margin-left: 3rem">
+      <summary style="list-style: none">Legend</summary>
+   </details>
+   <table style="border: 1px solid #000; height: 38px; border-left: none; border-top-right-radius: 4px; border-bottom-right-radius: 4px;">
+      <thead>
+         <tr>
+            <th style = "width:70px; height:34px; text-align: center; background-color: var(--missing-color); color: White">Absent</th>
+            <th style = "width:70px; height:34px; text-align: center; background-color: var(--present-color); color: White">Present</th>
+            <th style = "width:70px; height:34px; text-align: center; background-color: var(--excused-color)">Optional</th>
+            <th class="stripe-bg" style = "width:70px; height:34px; --color-a: var(--excused-color); --color-b: var(--partial-color); text-align: center; border-top-right-radius: 4px; border-bottom-right-radius: 4px;">Excused</th>
+         </tr>
+      </thead>
+   </table>
+   </div>
+      <div class = "cheesy-row" style="clear: right">
+        <div style="overflow-x: scroll; padding-bottom: 64px;" class = "column2">
+         <div>
+            <table style="table-layout: fixed; margin-bottom: 0px" class="table table-striped table-bordered table-condensed">
+               <tr>
+                  <% DB.fetch """
+                     WITH
+                        build_days AS (SELECT DISTINCT DATE(time_in) AS build_date FROM cheesy_frc_hours.lab_sessions)
+                     SELECT
+                        build_days.build_date,
+                        !ISNULL(cheesy_frc_hours.optional_builds.date) AS optional
+                     FROM
+                        build_days LEFT JOIN cheesy_frc_hours.optional_builds ON cheesy_frc_hours.optional_builds.date=build_date
+                     ORDER BY build_date ASC;
+                  """ do |build_date|%>
+                        <th style="width:60px; height:45px; padding-left: 0; padding-right: 0; text-align: center"> 
+                           <%= build_date[:build_date].strftime("%m/%d") %>
+                           <% if build_date[:optional] == 1 %>
+                              <br/>
+                              <a href="/delete_optional/<%=build_date[:build_date]%>"><small style="font-size: 0.65em">(Optional)</small></a>
+                           <% else %>
+                              <br/>
+                              <a href="/schedule_optional?date=<%=build_date[:build_date]%>" class="visible-if-parent-hover"><small style="font-size: 0.65em; padding: 0px 8px;">+optional</small></a>
+                           <% end %>
+                        </th>
+                  <% end %>
+               </tr>
+            </table>
+            <table style="table-layout: fixed;" class="table table-striped table-bordered table-condensed column-based">
+               <%studentsCnt = Student.count%>
+               <%rowNum = 1%>
+               <%DB.fetch """
```

Comment:
If you're going to have massive inline SQL queries like this, it should be either in `hours_server.rb` or here, but not both.